Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -307,19 +318,45 @@ static gboolean wpeDisplayDRMSetup(WPEDisplayDRM* displayDRM, const char* device
     displayDRM->priv->device = device;
     displayDRM->priv->connector = WTFMove(connector);
 
-    double scale;
-    if (const char* scaleString = getenv("WPE_DRM_SCALE"))
-        scale = g_ascii_strtod(scaleString, nullptr);
-    else {
-        auto* settings = wpe_display_get_settings(WPE_DISPLAY(displayDRM));
-        scale = wpe_settings_get_double(settings, WPE_SETTING_DRM_SCALE, nullptr);
+    static const auto scaleIsInBounds = [](double scale) {
+        return (scale >= 0.05) && (scale <= 20.0);
+    };
+
+    std::optional<double> scaleFromEnvironment;
+    if (const auto scaleString = StringView::fromLatin1(getenv("WPE_DRM_SCALE"))) {
+        RELEASE_ASSERT(scaleString.is8Bit());
+        auto trimmedScaleString = scaleString.trim(isASCIIWhitespace<LChar>);
+        size_t parsedLength = 0;
+        auto scale = parseDouble(trimmedScaleString, parsedLength);
+        if (parsedLength == trimmedScaleString.length() && scaleIsInBounds(scale))
+            scaleFromEnvironment = scale;
+        else
+            g_warning("Invalid WPE_DRM_SCALE='%*s' value, or out of bounds.", int(scaleString.span8().size()), scaleString.span8().data());
```

Comment:
```suggestion
            g_warning("Invalid WPE_DRM_SCALE='%*s' value, or out of bounds.", static_cast<int>(scaleString.span8().size()), scaleString.span8().data());
```

Reply:
Ah, this slipped. Sometimes to type less I just to `T()` for integral types, or even (gasp!) C casts. I'll change it to a `static_cast` before landing 👍🏼

Reply:
Done.